rfc/2022/2022-07-13 - Platform Wide Documentation vs Component Specific Documentation Locations.md

```diff
@@ -0,0 +1,75 @@
+
+## Motivation
+
+If we can make it easy for developers to locate and store technical documentation, we can save them hours (or even days) of extra work. Having a consistent method and location for finding this documentation is a crucial part of improving the experience and efficiency of VFS teams.
+
+## Design
+
+We propose that the central location for viewing component-specific technical documentation—including APIs, software components, and infrastructure—should be the Console catalog. The catalog makes it easy for users to navigate an org chart-type view of the VA ecosystem (including VFS apps, Platform services, and user groups) and quickly find documentation about any part of it.
+
+Using the Console catalog to solve the problems outlined above will necessarily also include the following:
+
+1. **A clear definition of what counts as “component-specific technical documentation.”** This definition will help teams know what documentation should be stored in the Console catalog versus on the Platform Website or in Confluence.
+    1. Documentation that should live in the Console catalog is:
+        1. specific to a software component, API/service, or piece of infrastructure
+        1. technical information needed for development
+        1. intended for use by only a subset of VA teams or roles
+    1. Examples include:
+        1. Documentation that is internal to a VFS team
+        1. Highly technical documentation about an infrastructure resource
+        1. Documentation for VA APIs and downstream external services
+1. **A clear relationship between the Platform Console and the Platform Website.** We propose that the Console and Website contain numerous references to each other. For example, if a component in the Console catalog has documentation that is relevant to all VFS users, it should have a link to Platform Website documentation on the topic. If the Platform Website has a page that mentions technical documentation that not all VFS teams need, it should link to the relevant documentation in the Console catalog. 
+1. **Storing documentation close to the software project/component.** Documentation for a  component should be stored in the same repo as the the component (where possible) or in a central documentation location, and then ingested into the Console catalog so it easy to find.
```

Comment:
This is very important and in line with the approach that my team is taking regarding where documentation should live. It seems like exposing these via the console will also help with overall discoverability of available documentation.

Reply:
@TPurple added https://github.com/department-of-veterans-affairs/va.gov-platform-architecture/pull/33/commits/3a9c5e2cd65ece69c086807489c5f66f469fbcf5 to highlight this point.